src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -86,6 +90,19 @@ private Optional<Set<Tag>> parseTagsForEdit(Collection<String> tags) throws Pars
         return Optional.of(ParserUtil.parseTags(tags));
     }
 
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     */
+    private Optional<Set<Tag>> parseTagsForRemoval(Collection<String> tags) throws ParseException {
+        assert tags != null;
+
+        if (tags.isEmpty() || tags.stream().allMatch(String::isEmpty)) {
+            return Optional.empty();
+        }
+        Collection<String> tagSet = tags.size() == 1 && tags.contains("") ? Collections.emptySet() : tags;
```

Comment:
Do we need this `tag.size() == 1 && tag.contains("")`? I thought it would already be covered by String::isEmpty.